pkg/monitors/http/http.go

```diff
@@ -0,0 +1,179 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/monitors"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+	"github.com/signalfx/signalfx-agent/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+func init() {
+	monitors.Register(&monitorMetadata, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `singleInstance:"false" acceptsEndpoints:"false"`
+	Body                 string            `yaml:"body"`
+	FollowRedirects      bool              `yaml:"followRedirects" default:"true"`
+	Headers              map[string]string `yaml:"headers"`
+	Method               string            `yaml:"method" default:"GET"`
+	Timeout              int               `yaml:"timeout" default:"5"`
+	URLs                 []string          `yaml:"urls"`
+	Regex                string            `yaml:"regex"`
+}
+
+// Monitor that collect metrics
+type Monitor struct {
+	Output types.FilteringOutput
+	cancel func()
+	logger logrus.FieldLogger
+}
+
+// Configure and kick off internal metric collection
+func (m *Monitor) Configure(conf *Config) error {
+	m.logger = logrus.WithFields(logrus.Fields{"monitorType": monitorType})
+	// Start the metric gathering process here
+	var ctx context.Context
+	ctx, m.cancel = context.WithCancel(context.Background())
+	utils.RunOnInterval(ctx, func() {
+		tlsCfg := &tls.Config{InsecureSkipVerify: true}
+		timeoutDuration := time.Duration(conf.Timeout) * time.Second
+		// add data
+		var body io.Reader
+		if conf.Body != "" {
+			body = strings.NewReader(conf.Body)
+		}
+		// get stats for each website
+		for _, site := range conf.URLs {
+			m.logger = m.logger.WithFields(logrus.Fields{"url": site})
```

Comment:
Feels a bit odd to be mutating m.logger each time. Could see wanting to run multiple URLs in parallel so probably best to avoid mutating shared state. Would make it a local variable.

Reply:
fixed